reconcile/vault_replication.py

```diff
@@ -0,0 +1,176 @@
+import logging
+import re
+
+from reconcile.utils.vault import VaultClient, SecretNotFound
+from reconcile.gql_definitions.jenkins_configs import jenkins_configs
+
+from reconcile.gql_definitions.vault_policies import vault_policies
+from reconcile.gql_definitions.vault_instances import vault_instances
+from reconcile.gql_definitions.vault_instances.vault_instances import (
+    VaultReplicationV1,
+)
+from reconcile.gql_definitions.vault_policies.vault_policies import (
+    VaultPolicyV1,
+)
+
+from reconcile.utils import gql
+
+from typing import List
+
+QONTRACT_INTEGRATION = "vault-replication"
+
+
+def copy_vault_secret(
+    dry_run: bool, source_vault: VaultClient, dest_vault: VaultClient, path: str
+) -> None:
+
+    secret_dict = {"path": path, "version": "LATEST"}
+    secrets, version = source_vault.read_all_version(secret_dict)
+    for secret in secrets:
+        write_dict = {"path": path, "data": secret}
+        try:
+            _, dest_version = dest_vault.read_all(secret_dict)
+            if dest_version < version:
+                logging.info(
+                    [
+                        "replicate_vault_secret",
+                        "Secret found, new version",
+                        version,
+                        path,
+                    ]
+                )
+                if not dry_run:
+                    dest_vault.write(write_dict)
+            else:
+                logging.info(["replicate_vault_secret", dest_version, version, secret])
+        except SecretNotFound:
+            logging.info(["replicate_vault_secret", "Secret not found", path])
+            if not dry_run:
+                dest_vault.write(write_dict)
+
+
+def copy_vault_secrets(
```

Comment:
can you change a bit the name of either this or the above function? too similar 😅

Reply:
Do we really need this function? It's just looping over the secrets and it's used only once. But if we do, please change the name as it is too close to `copy_vault_secret`

Reply:
Removed the function